boards/arm/xiao_ble/xiao_ble.dts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2022 Marcin Niestroj
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <nordic/nrf52840_qiaa.dtsi>
+#include "xiao_ble-pinctrl.dtsi"
```

Comment:
This should also include a seeeduino_xiao.dtsi that has the nexus node `xiao_d` and defines the labels `xiao_spi`, etc to make this a drop in board for a shield written against other XIAO compatible shields.

Reply:
I've added connector definitions based on ZMK source code.